### PR TITLE
Use correct python binary for Ubuntu 18 and EL-8.

### PR DESCRIPTION
**Pull request by akshay-joshi**

### CODE REVIEW

File: pkg/linux/build-functions.sh [hunk 1]
```diff
@@ -18,6 +18,19 @@ _setup_env() {
     if [ -n "${APP_SUFFIX}" ]; then
         APP_LONG_VERSION=${APP_LONG_VERSION}-${APP_SUFFIX}
     fi
+
+    # Setting up the correct Python version for Ubuntu 18 and EL-8, which have
+    # new Python versions installed parallel to the old ones.
+    OS_VERSION=$(grep "^VERSION_ID=" /etc/os-release | awk -F "=" '{ print $2 }' | sed 's/"//g' | awk -F "." '{ print $1 }')
+    SYSTEM_PYTHON_PATH='/usr/bin/python3'
+    PYTHON_BINARY='python3'
+    if [ "${OS_VERSION}" == 18 ]; then
```
Comment (dpage) [inline pkg/linux/build-functions.sh:27]: As this is in the shared functions file, we can't assume it's being run on Ubuntu or EL. The test should look more like:

if [ "$1" == "redhat" ] && [ "${OS_VERSION}" == "8" ]; then

Comment (dpage) [reply pkg/linux/build-functions.sh:27]: Additionally, you're comparing a string to an int. Quote the 18 :-)

File: pkg/debian/build.sh [hunk 2]
```diff
@@ -38,7 +37,7 @@ Version: ${APP_LONG_VERSION}
 Architecture: ${OS_ARCH}
 Section: database
 Priority: optional
-Depends: python3, libpq5 (>= 11.0), libgssapi-krb5-2
+Depends: ${PYTHON_BINARY}, libpq5 (>= 11.0), libgssapi-krb5-2
```
Comment (dpage) [inline pkg/debian/build.sh:40]: How safe is this, considering PYTHON_BINARY is being set in a child function?

File: pkg/redhat/setup.sh [hunk 3]
```diff
@@ -37,10 +37,10 @@ curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | sudo tee /etc/yu
 echo "Installing build pre-requisites..."
 yum groupinstall -y "Development Tools"
 
-if [ "${OS_VERSION}" == 7 ]; then
-    yum install -y expect fakeroot httpd-devel postgresql15-devel python3-devel nodejs yarn rpm-build rpm-sign yum-utils krb5-devel
-    pip3 install sphinx
-    pip3 install sphinxcontrib-youtube
+if [ "${OS_VERSION}" == 8 ]; then
```
Comment (dpage) [inline pkg/redhat/setup.sh:40]: This should also be properly quoted to avoid comparing a string and an int (I realise the old code was also broken).

**Review by dpage — CHANGES_REQUESTED** [review]

File: pkg/debian/build.sh (hunk 2 above)
Comment (akshay-joshi) [reply pkg/debian/build.sh:40]: We have set the PYTHON_BINARY  in the '**_setup_env**' function where other variables are set like {SERVERROOT}, {DISTROOT}, {APP_NAME} etc. I thought if those are being set, so PYTHON_BINARY should also be set.

Comment (dpage) [reply pkg/debian/build.sh:40]: Good point. No problem then.

**Pull request merged**